src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -306,23 +306,14 @@ string Bmi_Multi_Formulation::get_output_line_for_timestep(int timestep, std::st
         // This almost certainly should never happen, but just to be safe ...
         if (output_var_names.empty()) { return ""; }
 
-        try {
-            // Do the first separately, without the leading comma
-            auto output_data_provider_iter = availableData.find(output_var_names[0]);
-            *output_text_stream << get_var_value_as_double(output_var_names[0]);
-
-            // Do the rest with a leading comma
-            for (int i = 1; i < output_var_names.size(); ++i) {
-                *output_text_stream << "," << get_var_value_as_double(output_var_names[i]);
-            }
-            return output_text_stream->str();
-        }
-        catch (const std::exception &e) {
-            std::cerr << "WARN: " << e.what()
-                      << "; reverting to default behavior for multi-BMI formulation type (using last module)";
-            output_text_stream->str(std::string()); // ... clear any output contents being staged ...
-            is_out_vars_from_last_mod = true;          // ... revert to default behavior (just use last nested module)
+        // Do the first separately, without the leading comma
+        *output_text_stream << get_var_value_as_double(output_var_names[0]);
+
+        // Do the rest with a leading comma
+        for (int i = 1; i < output_var_names.size(); ++i) {
+            *output_text_stream << "," << get_var_value_as_double(output_var_names[i]);
```

Comment:
```suggestion
            *output_text_stream << delimiter << get_var_value_as_double(output_var_names[i]);
```
I just noticed an apparent bug/oversight in this function - the `delimiter` argument is just entirely unused.

At a broader level, it makes *no* sense to me that the BMI formulation class is doing string formatting of the variables being output, rather than just returning their values to be passed to an output routine.

Reply:
Yeah I am inferring that this goes back to the earliest prototypes. The mdframe PR #533 will help with this... however... we actually become memory-bound when running large simulations so storing the whole result in memory and dumping it out at the end will not be feasible, we'll have do develop some kind of streaming output system... which this at least accomplishes. That is probably why we're still doing this.